internal/controllers/topology/cluster/current_state.go

```diff
@@ -232,3 +264,44 @@ func (r *Reconciler) getCurrentMachineDeploymentState(ctx context.Context, clust
 	}
 	return state, nil
 }
+
+// alignRefAPIVersion returns an aligned copy of the currentRef so it matches the apiVersion in ClusterClass.
+// This is required so the topology controller can diff current and desired state objects of the same
+// version during reconcile.
+// If group or kind was changed in the ClusterClass, an exact copy of the currentRef is returned because
+// it will end up in a diff and a rollout anyway.
+// Only bootstrap template refs in a ClusterClass can change their group and kind.
```

Comment:
note, in theory there are also unsafe changes that can trigger this condition

Reply:
Are you referring to the annotation we talked about?

I couldn't fine it, only `unsafe.topology.cluster.x-k8s.io/disable-update-class-name-check` (which does something else)

Reply:
kk, I was wrong

Reply:
I'm also not sure how they resolved their issue at the time..